hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -90,54 +90,45 @@ namespace hazelcast {
                                     const std::shared_ptr<spi::impl::ClientInvocation> invocation) override {
                         check_connection(connection, invocation);
                         auto message = invocation->get_client_message();
-                        socket_strand_.post([=]() {
+                        socket_strand_.post([connection, invocation, message, this]() {
                             if (!check_connection(connection, invocation)) {
                                 return;
                             }
 
                             bool success;
                             int64_t message_call_id;
-                            do {
```

Comment:
I put this back. That loop is still needed for the very little possibility that the generated call ids clash. This can only happen usually if it is an event they stay in the map for a long time and if the counter overflows back to the original number.